docker-compose.yml

```diff
@@ -0,0 +1,189 @@
+version: '3.8'
+
+services:
+  angular-build:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
```

Comment:
After taking a full pass on the changes, I'm wondering if we can do the following:

Have two docker files one for the backend and one for the frontend, each downloading relevant dependencies and use YAML merge key ([ref](https://stackoverflow.com/questions/50072810/whats-the-double-arrow-django-mean-in-a-docker-compose-file)) wherever needed? DO you think this would be helpful for setup?

Reply:
Do we need to mention which stage to use for this build i.e, which FROM statement to use?

Reply:
yes, added that

Reply:
no, we cannot specify the build stage in the compose services. I tried doing that and then the docker action is failing : https://github.com/oppia/oppia/actions/runs/5500551767/jobs/10023536366
from the logs, the reasons seems that if we are using the target build then it is unable to add symlinks between the backend stage generated files and the FE stage generated files

Reply:
> it is unable to add symlinks between the backend stage generated files and the FE stage generated files

Why do we need to add a symlink between the backend and frontend? I think we need to look at this in detail and find the actual cause of the error.

Reply:
added build stages in the compose services. will be fixing the docker action asap